reporters/GlobalSummaryReporter.js

```diff
@@ -0,0 +1,476 @@
+
+class GlobalSummaryReporter {
+  constructor(
+    expiringMultiPartyClient,
+    expiringMultiPartyEventClient,
+    referencePriceFeed,
+    uniswapPriceFeed,
+    oracle,
+    collateralToken,
+    syntheticToken,
+    periodLengthSeconds
+  ) {
+    this.empClient = expiringMultiPartyClient;
+    this.empEventClient = expiringMultiPartyEventClient;
+    this.referencePriceFeed = referencePriceFeed;
+    this.uniswapPriceFeed = uniswapPriceFeed;
+
+    this.periodLengthSeconds = periodLengthSeconds;
+
+    this.web3 = this.empEventClient.web3;
+
+    this.empContract = this.empEventClient.emp;
+    this.collateralContract = collateralToken;
+    this.syntheticContract = syntheticToken;
+    this.oracleContract = oracle;
+
+    this.formatDecimalString = createFormatFunction(this.web3, 2, 4);
+  }
+
+  update = async () => {
+    const { toBN } = this.web3.utils;
+
+    await this.empClient.update();
+    await this.empEventClient.update();
+    await this.referencePriceFeed.update();
+    await this.uniswapPriceFeed.update();
+
+    // Block number stats.
+    this.currentBlockNumber = Number(await this.web3.eth.getBlockNumber());
+    this.startBlockNumberForPeriod =
+      this.currentBlockNumber - (await this._getLookbackTimeInBlocks(this.periodLengthSeconds));
+    this.periodLabelInHours = `${Math.round(this.periodLengthSeconds / (60 * 60))}H`;
+
+    // Events accessible by all methods.
+    this.collateralDepositEvents = await this.collateralContract.getPastEvents("Transfer", {
+      fromBlock: 0,
+      toBlock: this.currentBlockNumber,
+      filter: { to: this.empContract.options.address }
+    });
+    this.collateralWithdrawEvents = await this.collateralContract.getPastEvents("Transfer", {
+      fromBlock: 0,
+      toBlock: this.currentBlockNumber,
+      filter: { from: this.empContract.options.address }
+    });
+    this.syntheticBurnedEvents = await this.syntheticContract.getPastEvents("Transfer", {
+      fromBlock: 0,
+      toBlock: this.currentBlockNumber,
+      filter: { from: this.empContract.options.address, to: "0x0000000000000000000000000000000000000000" }
+    });
+    this.newSponsorEvents = this.empEventClient.getAllNewSponsorEvents();
+    this.createEvents = this.empEventClient.getAllCreateEvents();
+    this.regularFeeEvents = this.empEventClient.getAllRegularFeeEvents();
+    this.finalFeeEvents = this.empEventClient.getAllFinalFeeEvents();
+    this.liquidationEvents = this.empEventClient.getAllLiquidationEvents();
+    this.disputeEvents = this.empEventClient.getAllDisputeEvents();
+    this.disputeSettledEvents = this.empEventClient.getAllDisputeSettlementEvents();
+
+    // EMP Contract stats.
+    this.totalPositionCollateral = await this.empContract.methods.totalPositionCollateral().call();
+    this.totalTokensOutstanding = await this.empContract.methods.totalTokensOutstanding().call();
+    this.collateralLockedInLiquidations = toBN((await this.empContract.methods.pfc().call()).toString()).sub(
```

Comment:
+1 -- this is clever. Is the result of this `.call()` a BN already? Do we have to convert it to a string and back to a BN?

Reply:
`.call()` returns a (FixedPoint) object `[ '75166285203707845817894', rawValue: '75166285203707845817894' ]`, and if you `.toString()` that object it returns the rawValue

Reply:
Really? I thought calling a `.toString()` would print `[ '75166285203707845817894', rawValue: '75166285203707845817894' ]` or `[object Object]`? Why does it print out the rawValue here? Or is this that weird web3 behavior we noticed a while back?

Reply:
So I guess that would mean that `(await this.empContract.methods.pfc().call()).toString()` does the same thing as `(await this.empContract.methods.pfc().call()).rawValue()`, right?

Reply:
yeah not sure, see the results of logging out the object, the toString(), and the typeof https://github.com/UMAprotocol/protocol/pull/1548#discussion_r434169657

Reply:
> So I guess that would mean that `(await this.empContract.methods.pfc().call()).toString()` does the same thing as `(await this.empContract.methods.pfc().call()).rawValue`, right?
> 
> Not making any suggestions here -- just trying to wrap my head around it.

agreed